src/api/repository/user_repository.py

```diff
@@ -32,3 +32,12 @@ def check_db_connection():
         return "Connected"
     except Exception as e:
         return f"Failed to connect to database: {e}"
+
+class UserRepository:
+    def get_user_by_id(self, user_id: int):
+        fake_users = {
+            1: {'id': 1, 'name': 'Ann', 'email': 'Ann@example.com'},
+            2: {'id': 2, 'name': 'AJ', 'email': 'AJ@example.com'},
+        }
+        return fake_users.get(user_id)  # Returns None if user_id is not found
```

Comment:
Returning fake data for demo is okay, but the implementation should return the users from Database by getting the connection.

Use the psycopg connection to retrieve the user data from database.

Reply:
I have updated the code with below:

class UserRepository:
    def __init__(self, db_url: str):
        self.db_url = db_url 

    def get_connection(self):
        return psycopg.connect(self.db_url, row_factory=dict_row)

    def get_user_by_id(self, user_id: int):
        with self.get_connection() as conn:
            with conn.cursor() as cursor:
                cursor.execute("SELECT id, name, email FROM users WHERE id = %s", (user_id,))
                user = cursor.fetchone()  
                return user

Reply:
The above code looks ok for now, please push your changes and merge it after.

I approved the PR FYR.